gc/base/GCExtensionsBase.hpp

```diff
@@ -1093,6 +1101,22 @@ class MM_GCExtensionsBase : public MM_BaseVirtual {
 	MMINLINE void setConcurrentGlobalGCInProgress(bool inProgress) { _concurrentGlobalGCInProgress = inProgress; }
 #endif /* OMR_GC_MODRON_SCAVENGER */
 
+	/**
+	 * Determine whether Adaptive Threading is enabled. AdaptiveGCThreading flag
+	 * is not sufficient; Adaptive threading must be ignored if GC thread count is forced or concurrent
+	 * scavenger is enabled.
+	 * @return TRUE if adaptive threading routines can proceed, FALSE otherwise
+	 */
+	MMINLINE bool adaptiveThreadigEnabled()
+	{
+		bool adaptiveThreadingEnabled = (adaptiveGCThreading && !gcThreadCountForced);
+#if defined(OMR_GC_CONCURRENT_SCAVENGER)
+		return (!concurrentScavenger && adaptiveThreadingEnabled);
+#else /* defined(OMR_GC_CONCURRENT_SCAVENGER) */
+		return adaptiveThreadingEnabled;
+#endif /* defined(OMR_GC_CONCURRENT_SCAVENGER) */
```

Comment:
there is a helper isConcurrentScavengerEnabled() that already deals with ifdefs and can make this simpler